### PR TITLE
[Merged by Bors] - feat: `OfNat.ofNat n • x = n • x` and `Nat.cast n • x = n • x`

### PR DESCRIPTION
**Pull request by YaelDillies**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Command-Master — APPROVED** [review]

**Comment by mattrobball** [comment]
`algebraNat` and `algebraInt` should really be scoped. But as the library is, this is good. 

bors merge

File: Mathlib/Algebra/Algebra/Basic.lean [hunk 1]
```diff
@@ -330,7 +330,10 @@ theorem algebra_compatible_smul (r : R) (m : M) : r • m = (algebraMap R A) r 
 theorem algebraMap_smul (r : R) (m : M) : (algebraMap R A) r • m = r • m :=
   (algebra_compatible_smul A r m).symm
 
-theorem intCast_smul {k V : Type*} [CommRing k] [AddCommGroup V] [Module k V] (r : ℤ) (x : V) :
+lemma natCast_smul {k V : Type*} [Semiring k] [AddCommMonoid V] [Module k V] (n : ℕ) (x : V) :
+    (n : k) • x = n • x := algebraMap_smul k n x
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Basic.lean:334]: bors r-

We already have this lemma, its `nsmul_eq_smul_cast`. Can you deprecate the other one before creating this new one?

File: Mathlib/Algebra/Algebra/Basic.lean [hunk 2]
```diff
@@ -330,7 +330,10 @@ theorem algebra_compatible_smul (r : R) (m : M) : r • m = (algebraMap R A) r 
 theorem algebraMap_smul (r : R) (m : M) : (algebraMap R A) r • m = r • m :=
   (algebra_compatible_smul A r m).symm
 
-theorem intCast_smul {k V : Type*} [CommRing k] [AddCommGroup V] [Module k V] (r : ℤ) (x : V) :
+lemma natCast_smul {k V : Type*} [Semiring k] [AddCommMonoid V] [Module k V] (n : ℕ) (x : V) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Basic.lean:333]: How is this different to `natCast_smul_eq_nsmul`?

Comment (YaelDillies) [reply Mathlib/Algebra/Algebra/Basic.lean:333]: Sorry, I meant to delete these

**Comment by j-loreaux** [comment]
Thanks!

bors d+

File: Mathlib/Algebra/Module/Defs.lean [hunk 3]
```diff
@@ -326,18 +326,26 @@ section
...
 
 /-- `nsmul` is equal to any other module structure via a cast. -/
-theorem nsmul_eq_smul_cast (n : ℕ) (b : M) : n • b = (n : R) • b := by
+lemma natCast_smul_eq_nsmul (n : ℕ) (b : M) : (n : R) • b = n • b := by
   induction' n with n ih
   · rw [Nat.cast_zero, zero_smul, zero_smul]
   · rw [Nat.cast_succ, add_smul, add_smul, one_smul, ih, one_smul]
 
+/-- `nsmul` is equal to any other module structure via a cast. -/
+lemma ofNat_smul_eq_nsmul (n : ℕ) [n.AtLeastTwo] (b : M) : (OfNat.ofNat n : R) • b = n • b :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Defs.lean:335]: This should:
* Have `no_index` around the `ofNat`, and the usual library note
* Have `ofNat` on the RHS as well.

**Review by eric-wieser — APPROVED** [review]
Can you make a Zulip post about the new name? I think it's probably fine, but if we're going to create churn by changing it we may as well allow a day or two of bikeshedding.

File: Mathlib/Algebra/Module/Defs.lean (hunk 3 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Module/Defs.lean:335]: Yep, of course

**Comment by YaelDillies** [comment]
[Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/New.20name.20for.20Nat.2Ecast.20n.20.E2.80.A2.20x.20.3D.20n.20.E2.80.A2.20x)

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**